src/super_gradients/training/utils/quantization_utils.py

```diff
@@ -0,0 +1,384 @@
+     will be saved separately.
+
+    If performing calibration- the calibrated model is evaluated, and the metric_to_watch is logged under
+     calibrated_model_{metric_to_watch}. The calibrated checkpoint is saved under ckpt_calibrated_{calibration_method}.pth
+
+
+    Attributes:
+        start_epoch: int, first epoch to start QAT.
+
+        quant_modules_calib_method: str, One of [percentile, mse, entropy, max]. Statistics method for amax
+         computation of the quantized modules (default=percentile).
+
+        per_channel_quant_modules: bool, whether quant modules should be per channel (default=False).
+
+        calibrate: bool, whether to perfrom calibration (default=False).
+
+        calibrated_model_path: str, path to a calibrated checkpoint (default=None).
+
+        calib_data_loader: torch.utils.data.DataLoader, data loader of the calibration dataset. When None,
+         context.train_loader will be used (default=None).
+
+        num_calib_batches: int, number of batches to collect the statistics from.
+
+        percentile: float, percentile value to use when SgModel,quant_modules_calib_method='percentile'.
+         Discarded when other methods are used (Default=99.99).
+
+
+
+    """
+
+    def __init__(self, start_epoch: int, quant_modules_calib_method: str = "percentile",
+                 per_channel_quant_modules: bool = False, calibrate: bool = True, calibrated_model_path: str = None,
+                 calib_data_loader: DataLoader = None, num_calib_batches: int = 2, percentile: float = 99.99):
+        super(QATCallback, self).__init__(Phase.TRAIN_EPOCH_START)
+        self._validate_args(start_epoch, quant_modules_calib_method, calibrate, calibrated_model_path)
+        self.start_epoch = start_epoch
+        self.quant_modules_calib_method = quant_modules_calib_method
+        self.per_channel_quant_modules = per_channel_quant_modules
+        self.calibrate = calibrate
+        self.calibrated_model_path = calibrated_model_path
+        self.calib_data_loader = calib_data_loader
+        self.num_calib_batches = num_calib_batches
+        self.percentile = percentile
+
+    def _validate_args(self, start_epoch: int, quant_modules_calib_method: str, calibrate, calibrated_model_path):
+        if start_epoch < 0:
+            raise ValueError("start_epoch must be positive.")
+        if quant_modules_calib_method not in ["percentile", "mse", "entropy", "max"]:
+            raise ValueError(
+                "Unsupported quantization calibration method, expected one of: percentile, mse, entropy, max, got " + str(
+                    self.quant_modules_calib_method) + ".")
+        if not calibrate and calibrated_model_path is None:
+            logger.warning(
+                "calibrate=False and no calibrated_model_path is given. QAT will be on an uncalibrated model.")
+
+    def __call__(self, context: PhaseContext):
+        if context.epoch == self.start_epoch:
```

Comment:
`>=`

Reply:
Nope, this callback rebuilds the model so further training is QAT. This is triggered only at the start epoch.